quinn-proto/src/config.rs

```diff
@@ -308,25 +313,27 @@ where
 {
     /// Create a default config with a particular `reset_key`
     pub fn new(reset_key: S::HmacKey) -> Self {
+        let cid_factory: fn() -> Box<dyn ConnectionIdGenerator> =
+            || Box::new(RandomConnectionIdGenerator::default());
         Self {
-            local_cid_len: 8,
             reset_key: Arc::new(reset_key),
             max_udp_payload_size: MAX_UDP_PAYLOAD_SIZE,
+            connection_id_generator_factory: Arc::new(cid_factory),
         }
     }
 
-    /// Length of connection IDs for the endpoint.
+    /// assign a cid generator factory
     ///
-    /// This must be no greater than 20. If zero, incoming packets are mapped to connections only by
-    /// their source address. Otherwise, the connection ID field is used alone, allowing for source
-    /// address to change and for multiple connections from a single address. When local_cid_len >
-    /// 0, at most 3/4 * 2^(local_cid_len * 8) simultaneous connections can be supported.
-    pub fn local_cid_len(&mut self, value: usize) -> Result<&mut Self, ConfigError> {
-        if value > MAX_CID_SIZE {
-            return Err(ConfigError::OutOfBounds);
-        }
-        self.local_cid_len = value;
-        Ok(self)
+    ///
+    /// EndpointConfig::new() applies a default random CID generator factory.
+    /// This functions accepts any customized CID generator to reset CID generator factory that
+    /// implements ConnectionIdGenerator trait
+    pub fn set_cid_generator<F: Fn() -> Box<dyn ConnectionIdGenerator> + Send + Sync + 'static>(
```

Comment:
```suggestion
    pub fn cid_generator<F: Fn() -> Box<dyn ConnectionIdGenerator> + Send + Sync + 'static>(
```
For concision, we omit the `set_` prefix from methods on builders where ~all methods are setters.